infra/modules/providers/azure/service-principal/README.md

```diff
@@ -0,0 +1,45 @@
+# service-principal
+
+Manages a Service Principal within Azure Active Directory.
+
+- Optionally creates a new Service Principal and associated application in tenant of current subscription.
+- Assigns newly created Service Principal or a specified Service Principal to a role provided as a variable.
+
+## Usage
+
+New Service Principal usage example:
+
+```hcl
+
+module "service-principal" {
+  source                         = "../../modules/providers/azure/service-principal"
+  create_for_rbac                = "true"
+  service_principal_display_name = "TFTester"
+  scope                          = "/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333"
```

Comment:
May be mis-understanding, but I think this is `role_scope` based on `variables.tf`

Reply:
good catch